cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -110,36 +102,31 @@ public class AppFabricServer extends AbstractIdleService {
-                         DiscoveryService discoveryService,
-                         @Named(Constants.Service.MASTER_SERVICES_BIND_ADDRESS) InetAddress hostname,
-                         @Named(Constants.AppFabric.HANDLERS_BINDING) Set<HttpHandler> handlers,
-                         @Nullable MetricsCollectionService metricsCollectionService,
-                         ProgramRuntimeService programRuntimeService,
-                         RunRecordCorrectorService runRecordCorrectorService,
-                         ProgramRunStatusMonitorService programRunStatusMonitorService,
-                         ApplicationLifecycleService applicationLifecycleService,
-                         ProgramNotificationSubscriberService programNotificationSubscriberService,
-                         ProgramStopSubscriberService programStopSubscriberService,
-                         @Named("appfabric.services.names") Set<String> servicesNames,
-                         @Named("appfabric.handler.hooks") Set<String> handlerHookNames,
-                         AuditLogSubscriberService auditLogSubscriberService,
-                         CoreSchedulerService coreSchedulerService,
-                         CredentialProviderService credentialProviderService,
-                         NamespaceCredentialProviderService namespaceCredentialProviderService,
-                         ProvisioningService provisioningService,
-                         BootstrapService bootstrapService,
-                         SystemAppManagementService systemAppManagementService,
-                         TransactionRunner transactionRunner,
-                         RunRecordMonitorService runRecordCounterService,
-                         CommonNettyHttpServiceFactory commonNettyHttpServiceFactory,
-                         RunDataTimeToLiveService runDataTimeToLiveService,
-                         SourceControlOperationRunner sourceControlOperationRunner,
-                         RepositoryCleanupService repositoryCleanupService,
-                         OperationNotificationSubscriberService operationNotificationSubscriberService) {
+      DiscoveryService discoveryService,
+      @Named(Constants.Service.MASTER_SERVICES_BIND_ADDRESS) InetAddress hostname,
+      @Named(Constants.AppFabric.SERVER_HANDLERS_BINDING) Set<HttpHandler> handlers,
+      @Nullable MetricsCollectionService metricsCollectionService,
+      ProgramRuntimeService programRuntimeService,
+      ProgramRunStatusMonitorService programRunStatusMonitorService,
+      ApplicationLifecycleService applicationLifecycleService,
+      @Named("appfabric.services.names") Set<String> servicesNames,
+      @Named("appfabric.handler.hooks") Set<String> handlerHookNames,
+      CoreSchedulerService coreSchedulerService,
+      CredentialProviderService credentialProviderService,
+      NamespaceCredentialProviderService namespaceCredentialProviderService,
+      ProvisioningService provisioningService,
```

Comment:
Do you need this? Provisioning should happen in response to TMS events. Overall, I'd try to remove most of services here, leave handlers and then add services 1 by 1 to see which ones are needed (and why)

Reply:
ProvisioningService is needed in Appfabric Server for ProfileHttpHandler, ProvisionerHttpHandler.

ProvisioningService is needed in Appfabric Processor for ProgramNotificationSubscriberService, ProgramLifecycleService, RunRecordCorrectorService, RemoteExecutionTwillRunnerService.

Reply:
Created [CDAP-21111](https://cdap.atlassian.net/browse/CDAP-21111) for follow-up PR and added a `TODO` in `ProvisioningService.java`.